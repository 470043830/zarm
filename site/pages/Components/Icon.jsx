import Markdown from '@site/components/Markdown';

export default class Page extends Markdown {
  document = () => {
    return {
      document: require('@/components/icon/demo.md'),
      className: 'icon-page',
    };
  }
}
