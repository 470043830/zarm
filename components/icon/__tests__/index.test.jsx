import React from 'react';
import { render } from 'enzyme';
import toJson from 'enzyme-to-json';
import Icon from '../index';

describe('Icon', () => {
  it('renders all Icons', () => {
    const wrapper = render(
      <div>
        <Icon theme="primary" type="add" />
        <Icon theme="primary" type="arrow-left" />
        <Icon theme="primary" type="arrow-bottom" />
        <Icon theme="primary" type="arrow-right" />
        <Icon theme="primary" type="broadcast" />
        <Icon theme="primary" type="date" />
        <Icon theme="primary" type="deletekey" />
        <Icon theme="primary" type="info-round" />
        <Icon theme="primary" type="info-round-fill" />
        <Icon theme="primary" type="keyboard" />
        <Icon theme="primary" type="minus" />
        <Icon theme="primary" type="question-round" />
        <Icon theme="primary" type="required" />
        <Icon theme="primary" type="right-round-fill" />
        <Icon theme="primary" type="right-round" />
        <Icon theme="primary" type="right" />
        <Icon theme="primary" type="search" />
        <Icon theme="primary" type="time-circle" />
        <Icon theme="primary" type="warning-round-fill" />
        <Icon theme="primary" type="wrong-round-fill" />
        <Icon theme="primary" type="wrong-round" />
        <Icon theme="primary" type="wrong" />
      </div>,
    );
    expect(toJson(wrapper)).toMatchSnapshot();
  });

  it('renders Icons of different theme correctly', () => {
    const wrapper = render(
      <div>
        <Icon theme="primary" type="right" />
        <Icon theme="success" type="add" />
        <Icon theme="warning" type="info-round" />
        <Icon theme="danger" type="wrong-round-fill" />
      </div>,
    );
    expect(toJson(wrapper)).toMatchSnapshot();
  });

  it('renders Icons of different size correctly', () => {
    const wrapper = render(
      <div>
        <Icon size="lg" theme="primary" type="broadcast" />
        <Icon size="md" theme="primary" type="broadcast" />
        <Icon size="sm" theme="primary" type="broadcast" />
      </div>,
    );
    expect(toJson(wrapper)).toMatchSnapshot();
  });

  it('renders Icons using iconfont', () => {
    const MyIcon = Icon.createFromIconfont({
      scriptUrl: '//at.alicdn.com/t/font_64075_w7tvzqpf0n.js', // generated by iconfont.cn
    });
    const wrapper = render(
      <div>
        <MyIcon size="lg" theme="primary" type="icon-empty_fill" />
      </div>,
    );
    expect(toJson(wrapper)).toMatchSnapshot();
  });
});
