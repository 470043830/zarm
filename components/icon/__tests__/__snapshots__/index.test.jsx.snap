// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Icon renders Icons of different size correctly 1`] = `
<div>
  <i
    class="za-icon za-icon-broadcast za-icon--theme-primary za-icon--size-lg"
    theme="primary"
    type="broadcast"
  >
    <svg
      fill="currentColor"
      height="1em"
      viewBox="0 0 32 32"
      width="1em"
    >
      <use
        href="#broadcast"
      />
    </svg>
  </i>
  <i
    class="za-icon za-icon-broadcast za-icon--theme-primary za-icon--size-md"
    theme="primary"
    type="broadcast"
  >
    <svg
      fill="currentColor"
      height="1em"
      viewBox="0 0 32 32"
      width="1em"
    >
      <use
        href="#broadcast"
      />
    </svg>
  </i>
  <i
    class="za-icon za-icon-broadcast za-icon--theme-primary za-icon--size-sm"
    theme="primary"
    type="broadcast"
  >
    <svg
      fill="currentColor"
      height="1em"
      viewBox="0 0 32 32"
      width="1em"
    >
      <use
        href="#broadcast"
      />
    </svg>
  </i>
</div>
`;

exports[`Icon renders Icons of different theme correctly 1`] = `
<div>
  <i
    class="za-icon za-icon-right za-icon--theme-primary za-icon--size-md"
    theme="primary"
    type="right"
  >
    <svg
      fill="currentColor"
      height="1em"
      viewBox="0 0 32 32"
      width="1em"
    >
      <use
        href="#right"
      />
    </svg>
  </i>
  <i
    class="za-icon za-icon-add za-icon--theme-success za-icon--size-md"
    theme="success"
    type="add"
  >
    <svg
      fill="currentColor"
      height="1em"
      viewBox="0 0 32 32"
      width="1em"
    >
      <use
        href="#add"
      />
    </svg>
  </i>
  <i
    class="za-icon za-icon-info-round za-icon--theme-warning za-icon--size-md"
    theme="warning"
    type="info-round"
  >
    <svg
      fill="currentColor"
      height="1em"
      viewBox="0 0 32 32"
      width="1em"
    >
      <use
        href="#info-round"
      />
    </svg>
  </i>
  <i
    class="za-icon za-icon-wrong-round-fill za-icon--theme-danger za-icon--size-md"
    theme="danger"
    type="wrong-round-fill"
  >
    <svg
      fill="currentColor"
      height="1em"
      viewBox="0 0 32 32"
      width="1em"
    >
      <use
        href="#wrong-round-fill"
      />
    </svg>
  </i>
</div>
`;

exports[`Icon renders Icons using iconfont 1`] = `
<div>
  <i
    class="za-icon za-icon-icon-empty_fill za-icon--theme-primary za-icon--size-lg"
    theme="primary"
    type="icon-empty_fill"
  >
    <svg
      fill="currentColor"
      height="1em"
      viewBox="0 0 32 32"
      width="1em"
    >
      <use
        href="#icon-empty_fill"
      />
    </svg>
  </i>
  <i
    class="za-icon za-icon-icon-caidan07 za-icon--theme-primary za-icon--size-lg"
    theme="primary"
    type="icon-caidan07"
  >
    <svg
      fill="currentColor"
      height="1em"
      viewBox="0 0 32 32"
      width="1em"
    >
      <use
        href="#icon-caidan07"
      />
    </svg>
  </i>
</div>
`;

exports[`Icon renders Icons using svg component 1`] = `
<div>
  <i
    class="za-icon za-icon--theme-primary za-icon--size-lg"
    theme="primary"
    type=""
  >
    <svg
      fill="currentColor"
      height="1em"
      stroke="currentColor"
      viewBox="0 0 32 32"
      width="1em"
    >
      <path
        d="M1 12.376l8.8 9.114L30.431 1.568"
        fill="none"
        stroke="currentColor"
        stroke-width="2.6"
      />
    </svg>
  </i>
</div>
`;

exports[`Icon renders all Icons 1`] = `
<div>
  <i
    class="za-icon za-icon-add za-icon--theme-primary za-icon--size-md"
    theme="primary"
    type="add"
  >
    <svg
      fill="currentColor"
      height="1em"
      viewBox="0 0 32 32"
      width="1em"
    >
      <use
        href="#add"
      />
    </svg>
  </i>
  <i
    class="za-icon za-icon-arrow-left za-icon--theme-primary za-icon--size-md"
    theme="primary"
    type="arrow-left"
  >
    <svg
      fill="currentColor"
      height="1em"
      viewBox="0 0 32 32"
      width="1em"
    >
      <use
        href="#arrow-left"
      />
    </svg>
  </i>
  <i
    class="za-icon za-icon-arrow-bottom za-icon--theme-primary za-icon--size-md"
    theme="primary"
    type="arrow-bottom"
  >
    <svg
      fill="currentColor"
      height="1em"
      viewBox="0 0 32 32"
      width="1em"
    >
      <use
        href="#arrow-bottom"
      />
    </svg>
  </i>
  <i
    class="za-icon za-icon-arrow-right za-icon--theme-primary za-icon--size-md"
    theme="primary"
    type="arrow-right"
  >
    <svg
      fill="currentColor"
      height="1em"
      viewBox="0 0 32 32"
      width="1em"
    >
      <use
        href="#arrow-right"
      />
    </svg>
  </i>
  <i
    class="za-icon za-icon-broadcast za-icon--theme-primary za-icon--size-md"
    theme="primary"
    type="broadcast"
  >
    <svg
      fill="currentColor"
      height="1em"
      viewBox="0 0 32 32"
      width="1em"
    >
      <use
        href="#broadcast"
      />
    </svg>
  </i>
  <i
    class="za-icon za-icon-date za-icon--theme-primary za-icon--size-md"
    theme="primary"
    type="date"
  >
    <svg
      fill="currentColor"
      height="1em"
      viewBox="0 0 32 32"
      width="1em"
    >
      <use
        href="#date"
      />
    </svg>
  </i>
  <i
    class="za-icon za-icon-deletekey za-icon--theme-primary za-icon--size-md"
    theme="primary"
    type="deletekey"
  >
    <svg
      fill="currentColor"
      height="1em"
      viewBox="0 0 32 32"
      width="1em"
    >
      <use
        href="#deletekey"
      />
    </svg>
  </i>
  <i
    class="za-icon za-icon-info-round za-icon--theme-primary za-icon--size-md"
    theme="primary"
    type="info-round"
  >
    <svg
      fill="currentColor"
      height="1em"
      viewBox="0 0 32 32"
      width="1em"
    >
      <use
        href="#info-round"
      />
    </svg>
  </i>
  <i
    class="za-icon za-icon-info-round-fill za-icon--theme-primary za-icon--size-md"
    theme="primary"
    type="info-round-fill"
  >
    <svg
      fill="currentColor"
      height="1em"
      viewBox="0 0 32 32"
      width="1em"
    >
      <use
        href="#info-round-fill"
      />
    </svg>
  </i>
  <i
    class="za-icon za-icon-keyboard za-icon--theme-primary za-icon--size-md"
    theme="primary"
    type="keyboard"
  >
    <svg
      fill="currentColor"
      height="1em"
      viewBox="0 0 32 32"
      width="1em"
    >
      <use
        href="#keyboard"
      />
    </svg>
  </i>
  <i
    class="za-icon za-icon-minus za-icon--theme-primary za-icon--size-md"
    theme="primary"
    type="minus"
  >
    <svg
      fill="currentColor"
      height="1em"
      viewBox="0 0 32 32"
      width="1em"
    >
      <use
        href="#minus"
      />
    </svg>
  </i>
  <i
    class="za-icon za-icon-question-round za-icon--theme-primary za-icon--size-md"
    theme="primary"
    type="question-round"
  >
    <svg
      fill="currentColor"
      height="1em"
      viewBox="0 0 32 32"
      width="1em"
    >
      <use
        href="#question-round"
      />
    </svg>
  </i>
  <i
    class="za-icon za-icon-required za-icon--theme-primary za-icon--size-md"
    theme="primary"
    type="required"
  >
    <svg
      fill="currentColor"
      height="1em"
      viewBox="0 0 32 32"
      width="1em"
    >
      <use
        href="#required"
      />
    </svg>
  </i>
  <i
    class="za-icon za-icon-right-round-fill za-icon--theme-primary za-icon--size-md"
    theme="primary"
    type="right-round-fill"
  >
    <svg
      fill="currentColor"
      height="1em"
      viewBox="0 0 32 32"
      width="1em"
    >
      <use
        href="#right-round-fill"
      />
    </svg>
  </i>
  <i
    class="za-icon za-icon-right-round za-icon--theme-primary za-icon--size-md"
    theme="primary"
    type="right-round"
  >
    <svg
      fill="currentColor"
      height="1em"
      viewBox="0 0 32 32"
      width="1em"
    >
      <use
        href="#right-round"
      />
    </svg>
  </i>
  <i
    class="za-icon za-icon-right za-icon--theme-primary za-icon--size-md"
    theme="primary"
    type="right"
  >
    <svg
      fill="currentColor"
      height="1em"
      viewBox="0 0 32 32"
      width="1em"
    >
      <use
        href="#right"
      />
    </svg>
  </i>
  <i
    class="za-icon za-icon-search za-icon--theme-primary za-icon--size-md"
    theme="primary"
    type="search"
  >
    <svg
      fill="currentColor"
      height="1em"
      viewBox="0 0 32 32"
      width="1em"
    >
      <use
        href="#search"
      />
    </svg>
  </i>
  <i
    class="za-icon za-icon-time-circle za-icon--theme-primary za-icon--size-md"
    theme="primary"
    type="time-circle"
  >
    <svg
      fill="currentColor"
      height="1em"
      viewBox="0 0 32 32"
      width="1em"
    >
      <use
        href="#time-circle"
      />
    </svg>
  </i>
  <i
    class="za-icon za-icon-warning-round-fill za-icon--theme-primary za-icon--size-md"
    theme="primary"
    type="warning-round-fill"
  >
    <svg
      fill="currentColor"
      height="1em"
      viewBox="0 0 32 32"
      width="1em"
    >
      <use
        href="#warning-round-fill"
      />
    </svg>
  </i>
  <i
    class="za-icon za-icon-wrong-round-fill za-icon--theme-primary za-icon--size-md"
    theme="primary"
    type="wrong-round-fill"
  >
    <svg
      fill="currentColor"
      height="1em"
      viewBox="0 0 32 32"
      width="1em"
    >
      <use
        href="#wrong-round-fill"
      />
    </svg>
  </i>
  <i
    class="za-icon za-icon-wrong-round za-icon--theme-primary za-icon--size-md"
    theme="primary"
    type="wrong-round"
  >
    <svg
      fill="currentColor"
      height="1em"
      viewBox="0 0 32 32"
      width="1em"
    >
      <use
        href="#wrong-round"
      />
    </svg>
  </i>
  <i
    class="za-icon za-icon-wrong za-icon--theme-primary za-icon--size-md"
    theme="primary"
    type="wrong"
  >
    <svg
      fill="currentColor"
      height="1em"
      viewBox="0 0 32 32"
      width="1em"
    >
      <use
        href="#wrong"
      />
    </svg>
  </i>
</div>
`;
